### PR TITLE
test: fix RawConnectionDriver

### PR DESCRIPTION
**Pull request by lambdai**
To go with https://github.com/envoyproxy/envoy/pull/10998

We will introduce a small downtime when building from envoy-filter-example:master before https://github.com/envoyproxy/envoy/pull/10998 is merged.

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

### CODE REVIEW

**Comment by lambdai** [comment]
After set the envoy repo to local directory which reflects envoyproxy/envoy#10998
$ bazel test //:echo2_integration_test //http-filter-example:http_filter_integration_test //:envoy_binary_test
//:envoy_binary_test                                            (cached) PASSED in 2.1s
//:echo2_integration_test                                                PASSED in 2.1s
//http-filter-example:http_filter_integration_test                       PASSED in 32.9s

**Comment by lambdai** [comment]
It's not wise to push to master because  envoyproxy/envoy#10998 is not merged yet.

Could any one create a branch for me to push? I don't have the access rights. 
Envoyproxy needs a SHA in this repo. It doesn't have to be master branch.

File: echo2_integration_test.cc [hunk 1]
```diff
@@ -1,7 +1,42 @@
+#include <type_traits>
+
 #include "test/integration/integration.h"
 #include "test/integration/utility.h"
 
 namespace Envoy {
+namespace {
+
+// NoCreateConnectionDriver is used to provide "value" member regardless `createConnectionDriver`
```
Comment (mattklein123) [inline echo2_integration_test.cc:9]: This is too complicated. Please just remove all of this and put in the final code that you need. We can force merge it here, and then you can update the SHA of filter example in your other PR. Thank you!

Comment (lambdai) [reply echo2_integration_test.cc:9]: Sounds good. It's a great c++ template practice in the weekend though.

I am reverting this PR

**Comment by lambdai** [comment]
@mattklein123 reverted and the description of this PR is updated.

**Pull request merged**